README.md

```diff
@@ -82,6 +82,10 @@ make fmt
 make proto-gen
 ```
 
+### Package-specific documentation
+
+- [Shares](https://pkg.go.dev/github.com/celestiaorg/celestia-app/pkg/shares)
```

Comment:
[Non Blocking][Question]
Does this line help in publishing the docs to go doc, instead of writing the actual godoc? or it's for users?

Reply:
This line is just for users